src/App.tsx

```diff
@@ -17,18 +54,43 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                status={status}
+                setStatus={setStatus}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {
+                !isInitialized
+                  ? <Loader />
+                  : (
+                    <>
+                      <TodoList
+                        todos={visibleTodos}
+                        selectedTodoId={todoId}
+                        selectTodo={(id: number) => {
+                          setTodoId(id);
+                          setSelectedTodo(findSelectedTodo(id));
+                        }}
```

Comment:
You can just pass setSelectedTodo function
```suggestion
                        onTodoSelected={setSelectedTodo}
```